### PR TITLE
SERVER-2253 Vue Transition of Remote Config

### PR DESCRIPTION
**Pull request by abhisheksingla02**
Moved Json Editor to a global component.
Added components required for remote-config.

### CODE REVIEW

File: plugins/remote-config/frontend/public/templates/parameters.html [hunk 1]
```diff
@@ -12,14 +12,13 @@ <h2 class="cly-vue-remote-config__header"> {{i18n('remote-config.title')}}
     </template>
   </cly-header>
   <cly-main>
-    <json-editor></json-editor>
     <cly-datatable-n :rows="tableRows">
       <template v-slot:default="scope">
         <el-table-column type="expand" min-width="50" v-if="isDrillEnabled">
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/templates/parameters.html:17]: if drill is disabled,
will this code block show the parameters default value?

Comment (itsiprikshit) [reply plugins/remote-config/frontend/public/templates/parameters.html:17]: correct me if i am not wrong, due to this check, the table won't have an expandable column if drill is disabled.

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 2]
```diff
@@ -2,13 +2,332 @@
...
+                defaultTag: {
+                    value: 1,
+                    label: "#6C47FF"
+                },
+                colorTag: COLOR_TAG
+            };
+        },
+        computed: {
+            isOpen: function() {
+                return this.$store.getters["countlyRemoteConfig/parameters/showConditionDialog"];
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:199]: why add `showConditionDialog` in vuex?
won't keeping the dialogs show state in the components local state work?

File: plugins/remote-config/frontend/public/javascripts/countly.models.js [hunk 3]
```diff
@@ -122,6 +122,20 @@
...
+                            map[c._id] = {
+                                "condition": c.condition,
+                                "condition_color": c.condition_color,
+                                "condition_definition": c.condition_definition,
+                                "condition_name": c.condition_name,
+                                "seed_value": c.seed_value,
+                                "used_in_parameters": c.used_in_parameters
+                            };
+                        });
+                        context.dispatch("countlyRemoteConfig/parameters/conditionsMap", map, {root: true});
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.models.js:137]: as you must have noticed in the code, for conditions we have a separate sub module.
Therefore, actions, getters, mutations related to conditions should be within conditions submodule not inside parameters submodule.

Also why create a map for conditions when you already have conditions in the vuex in an array.
You can easily use .filter to get the relevant condition from the array.
If you want to use a map instead of array due to performance issues, I suggest you keep only the map then. You can get rid of the conditions array from the vuex.

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 4]
```diff
@@ -2,13 +2,332 @@
...
+            }
+        },
+        computed: {
+            conditions: function() {
+                var conditions = [];
+                var conditionsMap = this.$store.getters["countlyRemoteConfig/parameters/conditionsMap"];
+                var self = this;
+                if (this.parameter.conditions.length > 0) {
+                    this.parameter.conditions.forEach(function(condition) {
+                        var conditionProperties = conditionsMap[condition.condition_id];
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:80]: ahhh so here you fetching the condition from the conditionsMap.
I think you can use .filter on the conditions array that already exists in vuex.
That should be fine i guess.

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 5]
```diff
@@ -2,13 +2,332 @@
...
+                        conditions.push(ob);
+                    });
+                }
+                return conditions;
+            },
+            totalConditions: function() {
+                var total = 0;
+                if (this.parameter.conditions.length > 0) {
+                    this.parameter.conditions.forEach(function(condition) {
+                        total = total + condition.c ? condition.c : 0;
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:96]: What about adding the count of requests that received default value.
Shouldn't we add that too the total count in order to find the percentages?

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 6]
```diff
@@ -2,13 +2,332 @@
...
+                immediate: true,
+                handler: function(newValue) {
+                    this.currentVal = newValue;
+                }
+            },
+        },
+        methods: {
+            handleClose: function() {
+                this.$store.dispatch("countlyRemoteConfig/parameters/showJsonEditor", false);
+                this.$store.dispatch("countlyRemoteConfig/parameters/showJsonEditorForCondition", false);
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:142]: `showJsonEditor` and `showJsonEditorForCondition ` can be local states or event props with default values of this component.

Comment (itsiprikshit) [reply plugins/remote-config/frontend/public/javascripts/countly.views.js:142]: Reference to [this](https://github.com/Countly/countly-server/pull/2171/files?diff=unified&w=0#r759976037) and [this](https://github.com/Countly/countly-server/pull/2171/files?diff=unified&w=0#r759977398)

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 7]
```diff
@@ -2,13 +2,332 @@
...
+                            return ["cly.=", "cly.!=", "cly.contains", "cly.between", "cly.isset"].includes(operator.id);
+                        }
+                    }
+                })]
+            })];
+            additionalProperties.push(new countlyQueryBuilder.Property({
+                id: 'up.random_percentile',
+                name: CV.i18n("remote-config.conditions.random.percentile"),
+                type: countlyQueryBuilder.PropertyType.NUMBER,
+                group: 'User Properties',
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:172]: Localise this please.

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 8]
```diff
@@ -462,6 +791,7 @@
         if (countlyGlobal.plugins.indexOf("drill") > -1) {
             templates.push("/drill/templates/query.builder.v2.html");
         }
+        templates.push("/remote-config/templates/create-condition.html");
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:794]: shouldn't this also only load when drill is enable?

File: frontend/express/public/stylesheets/styles/blocks/_helpers.css [hunk 9]
```diff
@@ -0,0 +1 @@
+.cly-vue-tooltip-icon{font-size:20px;color:#a7aeb8;vertical-align:middle}.cly-in-page-notification{border-radius:4px;text-align:center;margin-bottom:32px}.cly-in-page-notification a{text-decoration:underline;cursor:hand}.cly-vue-back-link{cursor:pointer;color:#616161;font-size:12px;display:inline-block;text-transform:uppercase;margin-bottom:25px}.cly-vue-back-link:visited{cursor:pointer;color:#616161;font-size:12px;display:inline-block;text-transform:uppercase}.cly-vue-back-link:hover span{text-decoration:underline}.cly-divider{width:100%;border-top:1px solid #ececec}.cly-divider.is-2{border-top-width:2px}.cly-divider.is-dashed{border-top-style:dashed}.cly-divider.is-dotted{border-top-style:dotted}.cly-event-select .el-radio-group{display:flex}.cly-event-select .el-radio-button{flex-grow:1}.cly-event-select .el-radio-button__inner{width:100%}.cly-vue-color-tag__color-tag-wrapper{width:20%}.cly-vue-color-tag__color-tag{width:92%;height:37px;margin-bottom:5px;border-radius:4px}.cly-vue-color-tag__color-tag__selected{border:2px solid #0166d6}.cly-vue-color-tag__checkmark{color:#fff}/*# sourceMappingURL=_helpers.css.map */
```
Comment (itsiprikshit) [inline frontend/express/public/stylesheets/styles/blocks/_helpers.css:1]: remove this file. why is this added here?

File: frontend/express/public/stylesheets/styles/blocks/_helpers.css.map [hunk 10]
```diff
@@ -0,0 +1 @@
+{"version":3,"sourceRoot":"","sources":["_helpers.scss"],"names":[],"mappings":"AAGA,sBACI,eACA,cACA,sBAGJ,0BACE,kBACA,kBACA,mBAEA,4BACE,0BACA,YAGJ,mBACI,eACA,cACA,eACA,qBACA,yBACA,mBAEA,2BACI,eACA,cACA,eACA,qBACA,yBAIA,8BACI,0BAKZ,aACI,WACA,6BACA,kBACI,qBAEJ,uBACI,wBAEJ,uBACI,wBAKJ,kCACI,aAGJ,mCACI,YAEJ,0CACI,WAKJ,sCACI,UAGJ,8BACI,UACA,YACA,kBACA,kBACA,wCACI,yBAIR,8BACI","file":"_helpers.css"}
```
Comment (itsiprikshit) [inline frontend/express/public/stylesheets/styles/blocks/_helpers.css.map:1]: remove this file too. why is this added here?

File: plugins/remote-config/frontend/public/javascripts/countly.views.js [hunk 11]
```diff
@@ -2,13 +2,338 @@
...
+						{{i18n("remote-config.condition")}}\
+					</th>\
+                    		<th class="cly-vue-remote-config-percentages-breakdown__heading">\
+						{{i18n("remote-config.percentage")}}\
+					</th>\
+				</tr>\
+			</thead>\
+			<tbody>\
+                <tr><td><div class="cly-vue-remote-config-percentages-breakdown__sequence bu-py-1 bu-px-2">1</div></td><td><div class="bu-is-align-items-center	bu-is-flex cly-vue-remote-config-percentages-breakdown__data bu-p-1 cly-vue-remote-config-percentages-breakdown__default-value"><span class="bu-ml-2 bu-mr-3">{{i18n("remote-config.default-value")}}</span><span class="cly-vue-remote-config-percentages-breakdown__default-value__value bu-py-1 bu-px-2">{{defaultValue.value}}</span></div></td><td>{{defaultValue.percentage}} {{i18n("remote-config.percent.of.total")}}</td></tr>\
+				<tr v-if="isDrillEnabled" v-for="(condition, i) in conditions" :key="i">\
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/javascripts/countly.views.js:55]: does conditions array contain the default value aswell?

Comment (itsiprikshit) [reply plugins/remote-config/frontend/public/javascripts/countly.views.js:55]: if yes, aren't we supposed to show default value when drill is disabled? will default value be shown?

Comment (abhisheksingla02) [reply plugins/remote-config/frontend/public/javascripts/countly.views.js:55]: yes it will be shown :)

File: plugins/remote-config/frontend/public/templates/parameters.html [hunk 12]
```diff
@@ -12,14 +12,13 @@ <h2 class="cly-vue-remote-config__header"> {{i18n('remote-config.title')}}
...
   </cly-header>
   <cly-main>
-    <json-editor></json-editor>
     <cly-datatable-n :rows="tableRows">
       <template v-slot:default="scope">
-        <el-table-column type="expand" min-width="50" v-if="isDrillEnabled">
+        <el-table-column type="expand" min-width="50">
           <template v-slot="props">
             <cly-section>
               <el-tabs type="border-card" value="default_value">
```
Comment (itsiprikshit) [inline plugins/remote-config/frontend/public/templates/parameters.html:20]: i dont think there are tabs in expanded table form.
